catalyst-gateway/bin/src/db/event/common.rs

```diff
@@ -0,0 +1,27 @@
+//! Reusable common database objects
+
+#![allow(dead_code)]
+
+/// A query `LIMIT` and `OFFSET` limits.
+pub(crate) enum QueryLimits {
+    /// Return all entries without any `LIMIT` and `OFFSET` parameters
+    All,
+    /// Specifies `LIMIT` parameter equals to `1`
+    One,
```

Comment:
Is it worth having a special case for `One` when `Limit(1)` achieves the same thing?

Reply:
I think just for convenience and readability perspective yes.